caikit_nlp/modules/embedding_retrieval/embedding.py

```diff
@@ -0,0 +1,140 @@
+
+
+@module(
+    "EEB12558-B4FA-4F34-A9FD-3F5890E9CD3F",
+    "EmbeddingModule",
+    "0.0.1",
+    EmbeddingRetrievalTask,
+)
+class EmbeddingModule(ModuleBase):
+    def __init__(
+        self,
+        model: SentenceTransformer,
+    ):
+        super().__init__()
+        self.model = model
+
+    @classmethod
+    def load(cls, model_path: str) -> "EmbeddingModule":
+        """Load a sequence classification model
+
+        Args:
+            model_path: str
+                Path to the config dir of the model to be loaded.
+
+        Returns:
+            EmbeddingModule
+                Instance of this class built from the model.
+        """
+
+        config = ModuleConfig.load(model_path)
+        load_path = config.get("model_artifacts")
+
+        artifact_path = False
+        if load_path:
+            if os.path.isabs(load_path) and os.path.isdir(load_path):
+                artifact_path = load_path
+            else:
+                full_path = os.path.join(model_path, load_path)
+                if os.path.isdir(full_path):
+                    artifact_path = full_path
+
+        if not artifact_path:
+            artifact_path = config.get("hf_model", DEFAULT_HF_MODEL)
+
+        return cls.bootstrap(artifact_path)
+
+    def run(
+        self, input: List[str], **kwargs  # pylint: disable=redefined-builtin
+    ) -> EmbeddingResult:
+        """Run inference on model.
+        Args:
+            input: List[str]
+                Input text to be processed
+        Returns:
+            EmbeddingResult: the output
+        """
+
+        sentences: List[str]
+        if isinstance(input, str):
+            sentences = [input]
+        else:
+            sentences = input
+
+        result = self.model.encode(sentences)
+
+        vectors: List[Vector1D] = []
+        for vector in result:
+            vectors.append(Vector1D(vector))
+
+        return EmbeddingResult(vectors)
+
+    @classmethod
+    def bootstrap(cls, base_model_path: str) -> "EmbeddingModule":
+        """Bootstrap a HuggingFace transformer-based sequence classification model
+
+        Args:
+            base_model_path: str
+                Path to the model to be loaded.
+        """
+        model = SentenceTransformer(
+            base_model_path,
+            cache_folder=f"{HOME}/.cache/huggingface/sentence_transformers",
+        )
+        return cls(
+            model=model,
+        )
+
+    def save(self, model_path: str):
+        """Save model in target path
+
+        Args:
+            model_path: str
+                Path to store model artifact(s)
+        """
+        saver = ModuleSaver(
+            self,
+            model_path=model_path,
+        )
+
+        # Extract object to be saved
+        with saver:
+            saver.update_config({"model_artifacts": "."})
+            if self.model:  # This condition allows for empty placeholders
+                self.model.save(model_path)
```

Comment:
We would always want to save the model right? Whats the use-case to generate empty config object instead of model?

Reply:
Just fyi this is because we always edit the configs and don't use save.

The save/load test passed but I suppose the default model allowed it to pass when it should not have.

Reply:
Use case was to generate a config.yml for editing (e.g. add hf_model: model_name and run), but even without that this check avoids need for error check for self.model is None here when that responsibility is in other parts of the code.